### ISSUE TITLE
`pickVideo` requested additional permissions

### ISSUE THREAD

**Issue by fingerart** [flutter/flutter#146400]
### What package does this bug report belong to?

image_picker

### What target platforms are you seeing this bug on?

iOS

### Have you already upgraded your packages?

Yes

### Dependency versions

<details><summary>pubspec.lock</summary>

```lock
[Paste file content here]
```

</details>

### Steps to reproduce

I have consulted the documentation of `UIImagePickerController` and `PHPickerViewController`, and I can pull up the album selector without any permission. May I ask why `pickVideo` uses `checkCameraAuthorizationWithImagePicker` and `checkPhotoAuthorizationWithImagePicker` to request permissions by default?

1. https://discussionschinese.apple.com/thread/253834114?answerId=257189479322#257189479322
2. https://developer.apple.com/documentation/photokit/selecting_photos_and_videos_in_ios

https://github.com/flutter/packages/blob/0eead7cb335048092721c59d89936a985176b20d/packages/image_picker/image_picker_ios/ios/Classes/FLTImagePickerPlugin.m#L268-L275

### Expected results

No need to request additional permissions from the user.

### Actual results

Additional permissions requested.

### Code sample

<details open><summary>Code sample</summary>

```dart
None
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
None
```

</details>

**Comment by stuartmorgan** [comment]
This needs further investigation; I don't think it's the case that UIImagePickerController will operate in many cases without permissions. There are blame comments on some of the README about issues relating to crashes from missing permissions.

The new codepath is out-of-process, so doesn't need the same permissions.

**Comment by fingerart** [comment]
I have it working fine on two devices I have (iOS15 and iOS17) without asking for permission.
In addition, I think iOS14+ can use `PHPickerViewController`, and currently `UIImagePickerController` compressing video is also a problem. It is up to the developer to decide whether to compress the video or not. Another benefit of `PHPickerViewController` is that it does not present a cropped view to the user, more like a modern picker.

**Comment by hellohuanlin** [comment]
cc @vashworth to take a look

**Comment by fingerart** [comment]
Any progress?

**Comment by darshankawar** [comment]
@fingerart Please read [priority level](https://github.com/flutter/flutter/wiki/Issue-hygiene#priorities) details for your reference.

**Comment by vashworth** [comment]
So for `checkCameraAuthorizationWithImagePicker` -
* It is only requested if the `ImageSource` is `ImageSource.camera`. If we skip this check, it appears that iOS will automatically request the permissions (at least on iOS 16+, I can't guarantee on other versions). However, if the user rejects the permissions, we don't catch it and the user can actually continue to "record" and return a video (the video will be black). Whereas, with the `checkCameraAuthorizationWithImagePicker` in place, if the user rejects the permissions, it will throw an error that the user did not allow access.
* Also, `PHPickerViewController` can't be used for this use case as it does not have a capture functionality like `UIImagePickerController` does. 

For `checkPhotoAuthorizationWithImagePicker` - 
* This could be migrated to `PHPickerViewController` for iOS 14+. We already use it for the `getMedia` method (which allows you to pick image or videos). Upside to using `PHPickerViewController` is that it doesn't show the permissions dialogs, as stated in [WWDC 2020](https://developer.apple.com/videos/play/wwdc2020/10652%5D):
> [PHPicker] has privacy built in by default. The app doesn't need direct access to the photos library to show the picker, and the picker won't prompt for access on behalf of your app. It provides the user-selected photos and videos only, which is a clear user intent of sharing the assets with your app. You might wonder, "How do we achieve that?" Well, PHPicker runs out of process of the app even though it looks like PHPicker runs inside the app. It is really running in a separate process from the host app rendered on top of it.
* It does appear that `UIImagePickerController` has been [moved out of process too](https://developer.apple.com/documentation/photokit/selecting_photos_and_videos_in_ios#:~:text=Both%20PHPickerViewController%20and%20UIImagePickerController%20use%20the%20out%2Dof%2Dprocess%20Photos%20picker%20user%20interface.), and so the above statement _might_ also be true for it, but it's unclear if it doesn't require permissions anymore and if so, at what version. 

---
In summary, my findings include:

When using the camera to capture a video, `UIImagePickerController` and `checkCameraAuthorizationWithImagePicker` should be continued to be used as they are since there's not a `PHPickerViewController` equivalent and requesting permissions explicitly gives better feedback.

When selecting a video from the gallery, `PHPickerViewController` can be adopted for iOS 14+ and permissions don't need to be requested. However, for older iOS versions, `UIImagePickerController` and `checkPhotoAuthorizationWithImagePicker` should be continued to be used.

**Comment by fingerart** [comment]
@vashworth I agree with your conclusion. In addition, I have tested `PHPickerViewController`. Regardless of permissions, as long as the original video in the user's album has location information, the video file I obtained through `ffprobe` will also exist.

![img_v3_029o_f553b737-3df6-4515-9298-241f25aa841g 2](https://github.com/flutter/flutter/assets/6304197/6e970da0-5b2a-47c4-861b-75b54c3afc8f)

**Comment by idish** [comment]
Hi, currently, the `pickVideo` functionality is not modernized and basically unusable for production apps. There's really no good reason to use the old UIImagePickerController and requesting for extra permissions. Is there a fix on the roadmap?
Thanks!
